packages/flutter_tools/lib/src/commands/analyze_once.dart

```diff
@@ -48,6 +46,9 @@ class AnalyzeOnce extends AnalyzeBase {
 
   /// The working directory for testing analysis using dartanalyzer.
   final Directory workingDirectory;
+  final Completer<void> analysisCompleter = Completer<void>();
```

Comment:
I don't think these should be exposed as public fields. You could instead define `_handleAnalysisStatus` and `_handleAnalysisErrors` as closures and keep these objects above in the method body:

```

// Its OK to define methods with names in other method bodies.
void handleAnalysisStatus(bool isAnalyzing) {
    if (!isAnalyzing) {
      analysisCompleter.complete();
      subscription?.cancel();
      subscription = null;
    }
  }

try {
  subscription = server.onAnalyzing.listen((bool isAnalyzing) => handleAnalysisStatus(isAnalyzing));
```
Alternatively you could pass analysisCompleter/error as method arguments

Reply:
Get new knowledge Thanks